### PR TITLE
Deprecate v0::LSTMSequence

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Deprecate v0::LSTMSequence

### Tickets:
 - CVS-145457

### CODE REVIEW

**Comment by itikhono** [comment]
do we have plans to deprecate LSTMCell v0 as well?

**Comment by olpipi** [comment]
> do we have plans to deprecate LSTMCell v0 as well?

I suppose it will be the next step

**Comment by itikhono** [comment]
> > do we have plans to deprecate LSTMCell v0 as well?
> 
> I suppose it will be the next step

could you create a new ticket?

**Review by itikhono — APPROVED** [review]

**Review by mitruska — COMMENTED** [review]
Some checks rely on `ov::op::v0::LSTMSequence::get_type_info_static()`, 
does it also trigger a deprecation warning?
https://github.com/openvinotoolkit/openvino/blob/e7b0072c7ec04d4ae73eae4b2a2a2ad26550b524/src/plugins/intel_cpu/src/nodes/rnn.cpp#L251-L252

**Comment by olpipi** [comment]
> Some checks rely on `ov::op::v0::LSTMSequence::get_type_info_static()`, does it also trigger a deprecation warning?
> 
> https://github.com/openvinotoolkit/openvino/blob/e7b0072c7ec04d4ae73eae4b2a2a2ad26550b524/src/plugins/intel_cpu/src/nodes/rnn.cpp#L251-L252

It triggers warning only with msvc compiler

**Review by mitruska — APPROVED** [review]

File: src/core/include/openvino/op/lstm_sequence.hpp [hunk 1]
```diff
@@ -29,7 +29,9 @@ namespace v0 {
 ///
 ///
 /// \ingroup ov_ops_cpp_api
-class OPENVINO_API LSTMSequence : public util::RNNCellBase {
+class OPENVINO_API OPENVINO_DEPRECATED(
+    "The class ov::op::v0::LSTMSequence is deprecated. It will be removed in 2025.0 release. Use "
+    "ov::op::v5::LSTMSequence instead.") LSTMSequence : public util::RNNCellBase {
```
[the remark this answers is not in the document]
Comment (olpipi) [reply src/core/include/openvino/op/lstm_sequence.hpp:34]: done

**Review by p-durandin — APPROVED** [review]

**Comment by mitruska** [comment]
> I don't see why it shouldn't be fixed instead of deprecating. Could it be a regression? Do we have clear evidence that it is not used? The last time I proposed deprecating old ops it was rejected just because there was no clear evidence of value.

Based on the conformance tests v0::LSTMSequence is not supported by plugins.
Let's continue the dicsussion and clarify the concerns within the linked ticket.

**Comment by itikhono** [comment]
> I don't see why it shouldn't be fixed instead of deprecating. Could it be a regression? Do we have clear evidence that it is not used? The last time I proposed deprecating old ops it was rejected just because there was no clear evidence of value.

1. according to the conformance report, no plugins support v0 LSTM, instead of Template
2. only ONNX FE can produce v0 LSTM Sequence only in case of Peephole or input_forget modifications, but the inference will fails because of 1st point. It looks like we don't have any models in our scope using these LSTM modifications. Most probably it even never tested.
3. The new transformations do not cover v0 LSTM case
In general, supporting these v0 LSTM Seq/Cell operations will probably cost more effort than the value we can get from it.

**Comment by p-durandin** [comment]
@slyalin please consider reply @itikhono

**Pull request merged**